### PR TITLE
tests: Move job.go under libnet folder

### PR DESCRIPTION
**Pull request by oshoval**
**What this PR does / why we need it**:
`job.go` is used all by network
(one usage outside network folder is on migration.go for a subdomain FQDN test)
therefore move it under libnet folder.

Move `FormatIPForURL` as well to libnet.

Fix lint problems (it wasn't linted before because lint is done only on selected folders).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
None
```

### CODE REVIEW

File: tests/migration_test.go
Comment (oshoval) [inline tests/migration_test.go:628]: @EdDev 
are we ok with using network package here ?
otherwise we can / should move this specific Context (subdomain) to another file under network folder

Comment (EdDev) [reply tests/migration_test.go:628]: We should not import the tests into other tests.

Just move it to `libnet`.

**Review by EdDev — COMMENTED** [review]
Nice, thanks.

File: tests/migration_test.go
Comment (oshoval) [reply tests/migration_test.go]: Thanks, done
hopefully done correct, had to split like this in order to avoid cycle import

note that we can combine the two job.go files into one that exists on libnet if desired

File: tests/libnet/job/job.go
Comment (EdDev) [inline tests/libnet/job/job.go:1]: Please add the license header.

File: tests/network/job.go
Comment (EdDev) [inline tests/network/job.go:1]: Missing header license.

But I am not clear why this is not included under `libnet/job` as well.

File: tests/network/job.go
Comment (EdDev) [inline tests/network/job.go:49]: nit: Consider moving `FormatIPForURL` to `libnet` as well (different PR/commit).

**Review by EdDev — CHANGES_REQUESTED** [review]

File: tests/network/job.go
Comment (oshoval) [reply tests/network/job.go:1]: lets just merge the two whole job.go into the one that is on libnet/job ?
just split them because https://github.com/kubevirt/kubevirt/pull/10098#discussion_r1262404747 
but after doing that, it make more sense to merge them to one

Comment (EdDev) [reply tests/network/job.go:1]: I think it makes sense to merge them, both serve as helpers.

**Comment by oshoval** [comment]
Addressed resolved comments

File: tests/network/job.go
Comment (oshoval) [reply tests/network/job.go]: Done

File: tests/libnet/job/job.go
Comment (oshoval) [reply tests/libnet/job/job.go:20]: Maybe will be good to add presubmit that checks new files have a header

Comment (oshoval) [reply tests/libnet/job/job.go:20]: Please see https://github.com/kubevirt/kubevirt/pull/10119

**Review by EdDev — APPROVED** [review]
Thanks!

**Comment by oshoval** [comment]
Thanks

**Comment by oshoval** [comment]
`pull-kubevirt-code-lint` due to this PR, will take a look

**Comment by oshoval** [comment]
/hold

because this lane is optional

**Comment by EdDev** [comment]
> /hold cancel
> 
> Rebased Fixed lint problems (it seems it didnt run linter previously on that file, because it run on specific folders)

I cannot see the changes done because you rebased and fixed things.
Can you send links to the changes that are relevant?

**Comment by oshoval** [comment]
> > /hold cancel
> > Rebased Fixed lint problems (it seems it didnt run linter previously on that file, because it run on specific folders)
> 
> I cannot see the changes done because you rebased and fixed things. Can you send links to the changes that are relevant?

It was done in two separated pushed, but since it is in a commit, it appears different
please see the commit that appear in a line below https://github.com/kubevirt/kubevirt/pull/10098#issuecomment-1642173734

https://github.com/kubevirt/kubevirt/pull/10098/commits/11fcc686b19ee2e041f67066e38e084f7af1f9c3

**Pull request merged**
